src/doom/p_enemy.c

```diff
@@ -235,34 +235,26 @@ boolean P_CheckMissileRange (mobj_t* actor)
 
     dist >>= FRACBITS;
 
-    if (actor->type == MT_VILE)
-    {
-	if (dist > 14*64)	
-	    return false;	// too far away
-    }
-	
-
-    if (actor->type == MT_UNDEAD)
-    {
-	if (dist < 196)	
-	    return false;	// close for fist attack
-	dist >>= 1;
-    }
+    // generalization of the Arch Vile's different attack range
+    if (actor->info->maxattackrange > 0 && dist > actor->info->maxattackrange)
+	return false;	// too far away
 	
+    // generalization of the Revenant's different melee threshold
+    if (actor->info->meleethreshold > 0 && dist < actor->info->meleethreshold)
+	return false;	// close for fist attack
 
     if (actor->type == MT_CYBORG
 	|| actor->type == MT_SPIDER
+	|| actor->type == MT_UNDEAD
 	|| actor->type == MT_SKULL)
     {
 	dist >>= 1;
     }
     
-    if (dist > 200)
-	dist = 200;
-		
-    if (actor->type == MT_CYBORG && dist > 160)
-	dist = 160;
-		
+    // generalization of Min Missile Chance values hardcoded in vanilla
+    if (dist > actor->info->minmissilechance)
```

Comment:
`if (actor->info->minmissilechance > 0 &&` missing here?

Reply:
It wasn't missing, but I can add it.
Adding that, with ``minmissilechance == 0`` then the missile will be fired 100% of the time.
This would not affect any standard enemy (and even the empty mobj should set 200 for minmissilechance for them to have same behavior as they would before this change).